kombu/transport/virtual/base.py

```diff
@@ -462,9 +462,15 @@ def __init__(self, connection, **kwargs):
             typ: cls(self) for typ, cls in self.exchange_types.items()
         }
 
-        try:
-            self.channel_id = self.connection._avail_channel_ids.pop()
-        except IndexError:
+        # Cast to a set for fast lookups, and keep stored as an array for lower memory usage.
+        used_channel_ids = set(self.connection._used_channel_ids)
```

Comment:
Maybe we should put this `_used_channel_ids` stuff into its own method like it is in py-amqp?: https://github.com/celery/py-amqp/blob/6433be13199938f47bbc2826e6585e746baf4f9b/amqp/connection.py#L484-L492

Reply:
if that improves the codebase you could give it a try

Reply:
Ok, I went ahead and made this change: https://github.com/celery/kombu/pull/1470/commits/a439adaceae6c7e56e0d7c192ce63f2434a079e8